### PR TITLE
Reduced risk of reaching rate limit on github API

### PR DESCRIPTION
**Pull request by fdoxyz**
Implemented with `memory-cache`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -34,18 +35,38 @@ var octoberOpenPrs = [];
...
         user: username,
         page: pageNumber
-    }, function(err, res) {
+    };
+    if (cache.get('ETag?' + username + '&' + pageNumber) !== null) {
+        options.headers = {
+            "If-None-Match": cache.get('ETag?' + username + '&' + pageNumber)
+        };
+    }
+    console.log(options);
```
Comment (jenkoian) [inline index.js]: Remove debug.

File: index.js [hunk 2]
```diff
@@ -34,18 +35,38 @@ var octoberOpenPrs = [];
...
+    github.events.getFromUserPublic(options, function(err, res) {
 
         if (err) {
             deferred.reject();
         }
 
+        if (res.meta.status.substr(0,3) == "304") {
```
Comment (jenkoian) [inline index.js]: Triple = please

**Comment by jenkoian** [comment]
Dude this is fantastic, exactly what I had in mind, few minor things but looks great :)

**Comment by fdoxyz** [comment]
Awesome, it's actually my first PR, learned a thing or two in the process... Now it's time to get that counter running :P

**Pull request merged**